File: python/rosettacode_org/Find_limit_of_recursion/Find_limit_of_recursion.py

import sys


def set_recursion_limit():
    
    sys.setrecursionlimit(12345)

def defaut_recursion_limit():
    print(sys.getrecursionlimit())


def recurse(counter):
    print(counter)
    counter += 1
    recurse(counter)


#print(recurse(1))

def recurseDeeper(counter):
    try:
        print(counter)
        recurseDeeper(counter + 1)
    except RecursionError:
        print("RecursionError at depth", counter)
        recurseDeeper(counter + 1)

recurseDeeper(1)

File: python/rosettacode_org/Find_limit_of_recursion/Find_limit_of_recursion.js
function recurse(depth)
{
 try
 {
  return recurse(depth + 1);
 }
 catch(ex)
 {
  return depth;
 }
}
 
var maxRecursion = recurse(1);
console.log("Recursion depth on this system is " + maxRecursion);
//document.write("Recursion depth on this system is " + maxRecursion);
